### PR TITLE
11291 add opt out modal

### PR DESCRIPTION
**Pull request by rianfowler**
[Ticket](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/11291)

screen shot 2018-07-13 at 14 17 55

download-7

### CODE REVIEW

File: content/pages/education/gi-bill.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ majorlinks:
     - url: /education/gi-bill/survivors-dependent-assistance/
```
Comment (rianfowler) [inline content/pages/education/gi-bill.md]: these changes were pulled from [this](https://github.com/department-of-veterans-affairs/vets-website/pull/7947) content update

File: content/pages/education/index.md [hunk 2]
```diff
@@ -40,11 +40,16 @@ majorlinks:
     - url: /education/other-educational-assistance-programs/
```
Comment (rianfowler) [inline content/pages/education/index.md]: these changes were pulled from [this](https://github.com/department-of-veterans-affairs/vets-website/pull/7947) content update

File: content/pages/education/opt-out-information-sharing.md [hunk 3]
```diff
@@ -1,98 +1,54 @@
 ---
```
Comment (rianfowler) [inline content/pages/education/opt-out-information-sharing.md]: most of these changes were pulled from [this](https://github.com/department-of-veterans-affairs/vets-website/pull/7947) content update

File: content/pages/education/opt-out-information-sharing.md [hunk 4]
```diff
@@ -1,98 +1,54 @@
...
-collection: education
-order: 2
-relatedlinks:
-  - heading: Other VA benefits you may want to explore
-    links:
-    - url: /employment/vocational-rehab-and-employment/
-      title: Vocational Rehabilitation and Employment (VR&E)
-      description: If you have a service-connected disability that limits your ability to work or prevents you from working, find out if you can get VR&E benefits and services—like help exploring employment options and getting more training if required.
+order: 5
 widgets:
```
Comment (rianfowler) [inline content/pages/education/opt-out-information-sharing.md]: Adding a widget

File: content/pages/education/opt-out-information-sharing.md [hunk 5]
```diff
@@ -1,98 +1,54 @@
...
-<h4 itemprop="name">How long does it take VA to make a decision?</h4>
-<div itemprop="itemListElement">
-
-<div class="card information" markdown="0">
-<span class="number">30 days</span>
-<span class="description">Average time to process education claims</span>
-</div>
-</div>
-</div>
+<div id="react-applicationStatus" class="static-page-widget"></div>
```
Comment (rianfowler) [inline content/pages/education/opt-out-information-sharing.md]: Adds Opt out button at the bottom

File: src/applications/edu-benefits/components/OptOutWizard.jsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+          id="opt-out-alert"
+          cssClass="va-modal va-modal-large"
+          onClose={this.closeModal}
+          title="Are you sure you want to opt out?"
+          visible={this.state.modalOpen}>
+          <div>Here are some things that'll change if you ask VA to not share your education benefits information with schools:
+            <ul>
+              <li>You’ll be responsible for providing schools with your education benefits paperwork. If you transfer schools, you’ll also need to make sure your new school has your paperwork.</li>
+              <li>There may be a delay with you getting reimbursed for school-related expenses.</li>
+              <li>More information TK from stakeholders</li>
```
Comment (rianfowler) [inline src/applications/edu-benefits/components/OptOutWizard.jsx]: This content is still [pending](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/11291#issuecomment-401877057).

File: src/applications/edu-benefits/components/createOptOutApplicationStatus.jsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+
+const optOutForms = new Set(['22-0993']);
```
Comment (rianfowler) [inline src/applications/edu-benefits/components/createOptOutApplicationStatus.jsx]: I dunno if this is correct or necessary.

File: src/applications/edu-benefits/components/OptOutWizard.jsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+              <li>More information TK from stakeholders</li>
+            </ul>
+          </div>
+
+          <p><strong>Please note:</strong> If you submit an opt-out request and then change your mind, you’ll need to call the Education Call Center at 1-888-442-4551 to opt back in.</p>
+          <div>
+            <a
+              href="/education/opt-out-information-sharing/opt-out-form-0993"
```
Comment (rianfowler) [inline src/applications/edu-benefits/components/OptOutWizard.jsx]: This url is from this [comment](https://github.com/department-of-veterans-affairs/vets-website/pull/7975#issuecomment-403982841).

File: content/pages/education/index.md (hunk 2 above)
Comment (cehsu) [reply content/pages/education/index.md]: We're going to want to remove these cross link changes for now.

File: src/applications/edu-benefits/components/OptOutWizard.jsx [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+          <div>Here are some things that'll change if you ask VA to not share your education benefits information with schools:
+            <ul>
+              <li>You’ll be responsible for providing schools with your education benefits paperwork. If you transfer schools, you’ll also need to make sure your new school has your paperwork.</li>
+              <li>There may be a delay with you getting reimbursed for school-related expenses.</li>
+              <li>More information TK from stakeholders</li>
+              <li>More information TK from stakeholders</li>
+            </ul>
+          </div>
+
+          <p><strong>Please note:</strong> If you submit an opt-out request and then change your mind, you’ll need to call the Education Call Center at 1-888-442-4551 to opt back in.</p>
```
Comment (cehsu) [inline src/applications/edu-benefits/components/OptOutWizard.jsx]: This tel number should be formatted as an anchor according to the design, similar to as is done in the need help [component](https://github.com/department-of-veterans-affairs/vets-website/blob/43b510a621a31e73e51022661668047f84a808dd/src/applications/edu-benefits/components/GetFormHelp.jsx#L9): 
`        <a className="help-phone-number-link" href="tel:+1-888-442-4551">1-888-442-4551</a><br/>`

**Review by cehsu — APPROVED** [review]
LGTM pending a few minor changes, and @peggygannon's approval.

File: src/applications/edu-benefits/components/createOptOutApplicationStatus.jsx (hunk 7 above)
Comment (cehsu) [reply src/applications/edu-benefits/components/createOptOutApplicationStatus.jsx]: Interesting question, as there are multiple types of edu forms, the edu wizard supports showing a user multiple saved forms. It doesn't seem like this form should be considered an edu form in the sense that it should be included in that list, nor that there will be multiple opt-out forms.

File: src/applications/edu-benefits/components/OptOutWizard.jsx (hunk 9 above)
Comment (rianfowler) [reply src/applications/edu-benefits/components/OptOutWizard.jsx]: resolved.

File: src/applications/edu-benefits/components/createOptOutApplicationStatus.jsx (hunk 7 above)
Comment (rianfowler) [reply src/applications/edu-benefits/components/createOptOutApplicationStatus.jsx]: Makes sense- removing them didn't impact any behavior or cause any errors.

File: src/applications/edu-benefits/tests/components/OptOutWizard.unit.spec.jsx [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    component.update();
+    expect(component.state('modalOpen')).to.be.true;
+    expect(component.find('button').length).to.equal(3);
+    expect(component.find('a').length).to.equal(1);
+
+    delete global.document.focus;
+    global.document.querySelector = oldQuerySelector;
+  });
+  it('should close modal', () => {
+    // Modal uses query selector to get focus
```
Comment (rianfowler) [inline src/applications/edu-benefits/tests/components/OptOutWizard.unit.spec.jsx]: Modal is chaining some calls here that make it tricky to mock a SUT that uses it.

**Comment by rianfowler** [comment]
@peggygannon 

- This change has the WIP branch pulled in. I'll merge this after the content changes are merged into master.
- This includes the content changes to the modal.

**Pull request merged**